networks: activate connection properly and idempotancy for setup

SunilMohanAdapa opened the pull request:
When network-manager restarts, it leaves the existing connections intact. So, active the newly created connection to make them work.

Only add a connection for a device if a connection with that name is not already
associated with that device.

Closes #1045.

Minimal testing was done to ensure the results are better than what we have now.  May check again later with full disk image but I think the patch can be merged.

JosephKiranBabu commented:
@SunilMohanAdapa The "Closes"/"Fixes" keywords in Github automatically close issues on merge, only when they are part of the commit message. The pull request description is not considered.

SunilMohanAdapa commented:
I think trying to activate a connection in the above way will put us in trouble. If the network cable is not connected, and we configure that interface, we will try to start the connection, fail and will not continue to configure other interfaces.

SunilMohanAdapa commented:
Made a change to connection activation so that it happens in background and errors with activation are ignored.

Semi-tested the changes in a fresh disk image:
- Connections are activated.
- Setup does not fail when there is no network.

JosephKiranBabu commented:
@SunilMohanAdapa I built a virtualbox-amd64 image with this networks script and my changes for #1024.
The first image happens at boot time.
The second image is after installing gir1.2-nm-1.0 and starting plinth.
Since the background thread keeps running forever and the networks script keeps failing, the installation of essential modules doesn't complete and the user will be stuck on the screen shown in the third image.

![virtualbox_freedombox _06_10_2017_20_04_48](https://user-images.githubusercontent.com/4830191/31282963-c0b4d2ac-aad1-11e7-8090-acc53f73353d.png)
![virtualbox_freedombox _06_10_2017_20_00_37](https://user-images.githubusercontent.com/4830191/31282964-c0b53620-aad1-11e7-92ba-888668d1452e.png)
![screenshot from 2017-10-06 20-09-36](https://user-images.githubusercontent.com/4830191/31283117-5a460ed6-aad2-11e7-8e75-863d9212f0ca.png)

SunilMohanAdapa commented:
We have discovered that checking if the connection already exists is leading to failure in the following line:
```
already_exists=$(nmcli --terse --fields name,device con show | grep "$connection_name:$interface")
```
updating it to the following fixed the issue.
```
already_exists=$(nmcli --terse --fields name,device con show | grep "$connection_name:$interface" || true)
```

Updated the pull request accordingly and rebased onto the latest master.
